src/app/stylesheets/custom/button.less

```diff
@@ -42,15 +42,51 @@
     }
   }
 
+  &.btn-info {
+    color: @btn-text-color;
+    background-color: @btn-bg-color;
+    border-color: @btn-border-color;
+  }
+
+  &.btn-danger {
+    color: @btn-danger-text-color;
+    background-color: @btn-danger-bg-color;
+    border-color: @btn-danger-border-color;
+
+    &:hover {
+      background-color: darken(@btn-danger-bg-color, 10%);
+    }
+  }
+
+  &.btn-primary {
+    color: @btn-text-color;
+    background-color: @btn-bg-color;
+    border-color: darken(@btn-bg-color, 10%);
+
+    &:hover {
+      background-color: darken(@btn-bg-color, 10%);
+    }
+  }
+
+  &.btn-warning {
+    color: @btn-text-color;
+    background-color: @btn-bg-color;
+    border-color: @btn-border-color;
+
+    &:hover {
+      background-color: darken(@btn-bg-color, 10%);
+    }
+  }
+
   &.btn-shaded {
```

Comment:
I think `btn-shaded` should be replaced by the default button `btn-default`.

Primary actions should use `btn-primary`. As destroy action should use `btn-warning` or `btn-danger`.